exporter/kafkaexporter/pdata_marshaler.go

```diff
@@ -82,17 +85,30 @@ type pdataTracesMarshaler struct {
 	encoding  string
 }
 
-func (p pdataTracesMarshaler) Marshal(td pdata.Traces, topic string) ([]*sarama.ProducerMessage, error) {
-	bts, err := p.marshaler.MarshalTraces(td)
-	if err != nil {
-		return nil, err
-	}
-	return []*sarama.ProducerMessage{
-		{
+func (p pdataTracesMarshaler) Marshal(td pdata.Traces, topic string) (messages []*sarama.ProducerMessage, errs error) {
+	for _, batch := range batchpersignal.SplitTraces(td) {
+		bts, err := p.marshaler.MarshalTraces(td)
+		if err != nil {
+			// continue to process batches that can be serialized
+			errs = multierr.Append(errs, err)
+			continue
+		}
+
+		var key sarama.Encoder
+		if batch.ResourceSpans().Len() > 0 &&
```

Comment:
The if statement of fun... I've tried to rewrite this to see if I could make it more readable and unfortunately I can not.

I did think that was a top level check to examine all values internally but I couldn't see it listed anymore...